src/scheduler/pbs_sched.c

```diff
@@ -616,48 +617,54 @@ connect_server_helper(char *host, int is_primary)
 static void
 connect_server(char *svrhost)
 {
-	int primary_fd = -1;
-	int secondary_fd = -1;
-	svr_t *svr;
+	int primary_sock = -1;
+	int secondary_sock = -1;
+	sched_svrconn *svr;
+	int i;
 
 	if (!svrhost)
-		svrhost = pbs_default();
+		return;
 
-	while (primary_fd < 0 || primary_fd < 0) {
-		if (primary_fd < 0) {
-			primary_fd = connect_server_helper(svrhost, 1);
-			if (primary_fd < 0) {
+	while (primary_sock < 0 || secondary_sock < 0) {
+		if (primary_sock < 0) {
+			primary_sock = connect_server_helper(svrhost, 1);
+			if (primary_sock < 0) {
 				/* wait for 2s for not to burn too much CPU, and then retry connection */
 				sleep(2);
 				continue;
 			}
 		}
-		secondary_fd = connect_server_helper(svrhost, 0);
-		if (secondary_fd < 0) {
+		secondary_sock = connect_server_helper(svrhost, 0);
+		if (secondary_sock < 0) {
 			/* wait for 2s for not to burn too much CPU, and then retry connection */
 			sleep(2);
 			continue;
 		}
 	}
-	if (tpp_em_add_fd(poll_context, secondary_fd, EM_IN | EM_HUP | EM_ERR) < 0) {
+	if (tpp_em_add_fd(poll_context, secondary_sock, EM_IN | EM_HUP | EM_ERR) < 0) {
 		log_errf(-1, __func__, "Couldn't add secondary connection to poll list for server %s", svrhost);
-		pbs_disconnect(primary_fd);
-		pbs_disconnect(secondary_fd);
+		pbs_disconnect(primary_sock);
+		pbs_disconnect(secondary_sock);
 		die(-1);
 	}
 
-	svr = malloc(sizeof(svr_t));
+	svr = malloc(sizeof(sched_svrconn));
 	if (!svr) {
-		log_errf(-1, __func__, "Out of memory while allocating svr struct");
+		log_errf(-1, __func__, MEM_ERR_MSG);
 		die(-1);
 	}
 
-	svr->svrhost = svrhost;
-	svr->primary_fd = primary_fd;
-	svr->secondary_fd = secondary_fd;
-	CLEAR_LINK(svr->all_svrs_link);
-	append_link(&servers, &svr->all_svrs_link, svr);
-	set_validate_sched_attrs(svr->primary_fd);
+	svr->svrhost = strdup(svrhost);
+	if (!svr->svrhost) {
+		free(svr);
+		log_errf(-1, __func__, MEM_ERR_MSG);
+		die(-1);
+	}
+	svr->primary_sock = primary_sock;
+	svr->secondary_sock = secondary_sock;
+	i = 0;
+	while (servers[i] != NULL) i++;
+	servers[i] = svr;
```

Comment:
Consider using add_ptr_to_array()

Reply:
Thanks, I was looking for func like that in sched code, but couldn't find it, as add_* is in server_info.c/.h and remove_* is in misc.c/.h, will change to use add_ptr_to_array()

Reply:
Oh, that sounds like a mistake.  The add and remove functions should both be in misc.c.  I'll have to move it at some point.  Unless you want to do it.  Don't feel obligated to, though.

Reply:
Sure, will move it, (assuming) its just copy paste from one file to another.